packages/core/src/index.ts

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/require-await */
+
+export enum RouteType {
+  GMP = "gmp",
+  FUNGIBLE = "fungible",
+}
+
+export enum Environment {
+  LOCAL = "local",
+  DEVNET = "devnet",
+  TESTNET = "testnet",
+  MAINNET = "mainnet",
+}
+
+export enum FeeHandlerType {
+  BASIC = "basic",
+  PERCENTAGE = "percentage",
+}
+
+export enum Network {
+  EVM = "evm",
+  SUBSTRATE = "substrate",
+}
+
+export enum SecurityModel {
+  MPC = "mpc",
+}
+
+export type Domain = {
+  sygmaId: number;
+  // https://chainagnostic.org/CAIPs/caip-2
+  caipId: string;
+  chainId: number;
+  name: string;
+  iconUrl?: string;
+  type: Network;
+};
+
+export type Resource = EvmResource | SubstrateResource;
+
+export enum ResourceType {
+  FUNGIBLE = "fungible",
+  NON_FUNGIBLE = "nonfungible",
+  PERMISSIONED_GENERIC = "permissionedGeneric",
+  PERMISSIONLESS_GENERIC = "permissionlessGeneric",
+}
+
+interface BaseResource {
+  sygmaResourceId: string;
+  // https://chainagnostic.org/CAIPs/caip-19
+  caip19: string;
+  type: ResourceType;
+  iconUrl?: string;
+  native?: boolean;
+  burnable?: boolean;
+  symbol?: string;
+  decimals?: number;
+}
+
+export type EvmResource = BaseResource & {
+  address: string;
+};
+
+export type XcmMultiAssetIdType = {
+  concrete: {
+    parents: number;
+    interior:
+      | "here"
+      | {
+          x3: Array<{ parachain: number } | { generalKey: [number, string] }>; // This is a tuple of length and value
+        };
+  };
+};
+
+export type SubstrateResource = BaseResource & {
+  assetID?: number;
+  assetName: string;
+  xcmMultiAssetId: XcmMultiAssetIdType;
+};
+
+export type Route = {
+  fromDomain: Domain;
+  toDomain: Domain;
+  resource: Resource;
+  feeHandler: FeeHandler;
```

Comment:
#363 ^^

Reply:
After having a discussion with @saadjhk we believe having `feeHandler` doesn't make sense here. 
This would mean that the SDK for getRoutes would need to call N*2 RPC calls (where N is a number of routes).

This doesn't mean we should skip #363, as we should remove this dependency inside the `getFee` method - where SDK would do the on-chain query for this information.

@mpetrunic wdyt?

Reply:
@mpetrunic for each route fetching the `_domainResourceIDToFeeHandlerAddress` and then the type of `feeHandler`

do you want to show the fee type e.g percentage with each route and the percentage value as well

Reply:
ideally yes

Reply:
but if you think it's gonna do too much requests or be too slow, we could extract into separate method, let's say `getRouteFeeDetails`